docs/signature.md

```diff
@@ -0,0 +1,93 @@
+# Image Signature Specification
+
+**Version 0.1**
+
+## Introduction
+
+This document defines a detached container image signature object and signing methods.
+
+## Signature Format
+
+```js
+{
+    "critical": {/* required fields */
+        "identity": {/* identity reference */},
+        "image": {/* signed object reference */ },
+        "type": "..."
+    },
+    "optional": {/* optional metadata fields */}
+    }
+}
+```
+
+### Fields
+
+There are two top-level fields, **critical** (required) and **optional** (optional).
+
+#### `critical`
+
+**identity** (string):
+
+```js
+{
+    "docker-reference": imageName
+}
+```
+
+`imageName` per [V2 API](https://docs.docker.com/registry/spec/api/#/overview) Required.
```

Comment:
> You can avoid this by not including transport/location information in the asserted name.

That doesn’t really work; we need a global namespace for looking up the trusted key (cryptographic signature verification should happen before anything even _touches_ the JSON data!), and we need to be able to compare the name inside the signature with something the user has provided which unambiguously specifies intent. The only really practical way to do this is to tie both the key lookup and the name comparison to the name/reference/“pull specification” used by the user when saying “get/start this container”, because that’s the _only_ identification which we will always have without requiring the user to change their workflow.

It is acceptable to ask the user to do an one-time setup to trust key X for example.com; it is completely unreasonable to ask the user to say `$tool pull --expected-name='Product By Example.COM 5.1.3' example.com/product5/standalone:1.3`.